### DISCUSSION TITLE
bpf_btf_info has no member named 'name'

### DISCUSSION THREAD

**Discussion by baiwfg2** [bpftrace/bpftrace#2694] [Q&A]
I build v0.18.0 (already built bcc & libbpf, so use `-DUSE_SYSTEM_BPF_BCC=1`) and encounter the following issue.
```
src/bpffeature.cpp:565: error struct bpf_btf_info has no member named 'name'
```

I dig into it and find my kernel is 4.19.36, so it has no name field within struct bpf_btf_info.

Does it mean I cannot build v0.18.0 on those kernel that doesn't support name field within struct bpf_btf_info ? 

However I also find in libbpf submodule there's `include/uapi/linux/bpf.h` used by libbpf compiling and not installed by libbpf installation. So can they be used by bpftrace building ?

**Comment by viktormalik** [comment]
Try `make install_uapi_headers` from libbpf, that should install the necessary header.

**Reply by baiwfg2** [reply]
Thanks. Worked

**Reply by V0idk** [reply]
...how to find that

**Reply by danobi** [reply]
@V0idk https://github.com/libbpf/libbpf/blob/20c0a9e3d7e7d4aeb283eae982543c9cacc29477/src/Makefile#L163-L166

But if you're building as an end user I'd recommend just using the nix build: https://github.com/iovisor/bpftrace/blob/master/docs/developers.md#nix-build . It's basically guaranteed to work.
